server/src/domain/system-config/dto/system-config-oauth.dto.ts

```diff
@@ -23,6 +23,11 @@ export class SystemConfigOAuthDto {
   @IsString()
   clientSecret!: string;
 
+  @ValidateIf(isEnabled)
+  @IsNotEmpty()
```

Comment:
This validation isn't needed imo. It should always be required and 0 should indicate unlimited quota.